### DISCUSSION TITLE
better WebGL Performances with version 10.1.0

### DISCUSSION THREAD

**Discussion by obiot** [melonjs/melonJS#1061] [Announcements]
as part of the just released 10.1.0 version (see [here](https://github.com/melonjs/melonJS/releases/tag/10.1.0) for Github or [here](https://www.npmjs.com/package/melonjs/v/10.1.0) on NPM) we managed to squeeze in a few performances improvements on our in-house WebGL renderer.

* First of all the WebGL Compositor was further refactored, and while this one might not really make a difference, the simplified API will enable more improvements in the near future and make the compositor more generic and easier to customize.
* Temporary array allocation, when copying the vertex buffer data, has been improved (or shall I say removed) by leveraging the new WebGL2 API, as the [`gl.bufferData()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferData) method now allows to specify a custom starting and ending offset for the vertex buffer (while with WebGL1 a subset was manually created using `subArray()` every time the Framebuffer was flushed).
* Tint information is now passed as a single packed Uint32 to the vertex shader, as opposed to passing a 4 entries Float32Array for every texture, multiplying this by the amount of sprite or draw operations in your scene, that's a lots of data we do not need to pass to the GPU every time !

Let us know what kind of improvements you noticed in your game, and if you managed to also squeeze in more sprites or added complexity in your scene graph with the latest version :)

**Comment by obiot** [comment]
@parasyte I thought you would like this one :)

**Reply by parasyte** [reply]
Awesome! Good work, and congrats on the melonJS 2 release (version 10, hehe).
